src/core/media/hybris_recorder_observer.cpp

```diff
@@ -24,20 +24,12 @@ namespace media = core::ubuntu::media;
 
 struct media::HybrisRecorderObserver::Private
 {
-    struct Holder
-    {
-        std::weak_ptr<media::HybrisRecorderObserver::Private> wp;
-    };
-
     // The fringe that we hand over to hybris.
     static void on_media_recording_state_changed(bool started, void* context)
     {
-        if (auto holder = static_cast<Holder*>(context))
+        if (auto thiz = static_cast<HybrisRecorderObserver*>(context))
```

Comment:
Yes, I've now tried it and I verified that *sometimes* the callback is not invoked. However, I doubt that this is due to my changes, since I added some debugging messages and verified that it's the platform API who is not invoking the callback. So, I suspect that the same issue must have been present even in the old implementation, and that you were just lucky to not reproduce it there :-)